src/soundsourceproxy.cpp

```diff
@@ -219,24 +170,18 @@ void SoundSourceProxy::loadPlugins() {
     s_soundSourceProviders.registerProvider(Mixxx::SoundSourceProviderPointer(
             new Mixxx::SoundSourceProviderCoreAudio));
 #endif
-#ifdef __FFMPEGFILE__
-    // FFmpeg currently overrides all other built-in providers
-    // if enabled
-    s_soundSourceProviders.registerProvider(Mixxx::SoundSourceProviderPointer(
-            new Mixxx::SoundSourceProviderFFmpeg));
-#endif
 
     // Scan for and initialize all plugins.
     // Loaded plugins will replace any built-in providers
     // that have been registered before (see above)!
     const QList<QDir> pluginDirs(getSoundSourcePluginDirectories());
-    foreach (QDir dir, pluginDirs) {
-        qDebug() << "Loading SoundSource plugins" << dir.path();
-        const QStringList files(dir.entryList(
+    for (const auto& pluginDir: pluginDirs) {
```

Comment:
I think the cooperative approach is sufficient for the moment.

I've added functionality to deregister a provider for a specific file extension together with a function to (re-)register a provider for a specific file extensions with a custom priority. These 2 functions can be used to change the default priority of a provider for a specific file extension.

Modifying the provider registry at runtime might require additional locking, but all the required functionality will already be available.